packages/console/src/components/MultiTextInput/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useMemo } from 'react';
+import { FieldPath, FieldValues, RegisterOptions, UseFormRegister } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import IconButton from '@/components/IconButton';
+import * as textButtonStyles from '@/components/TextButton/index.module.scss';
+import TextInput from '@/components/TextInput';
+import Minus from '@/icons/Minus';
+
+import * as styles from './index.module.scss';
+
+export type Props<TFieldValues extends FieldValues = FieldValues> = {
+  watchedValue?: string[];
+  onChange: (value: string[]) => void;
+  register: UseFormRegister<TFieldValues>;
+  registerNameGenerator: (index: number) => FieldPath<TFieldValues>;
```

Comment:
What about just using a name props for the MultiTextInput field, and ${name}_${index} for all the input names inside?